event_monitor/src/lib.rs

```diff
@@ -16,8 +16,10 @@ static mut MONITOR: Option<(File, Instant)> = None;
 /// This function must only be called once from the main process before any threads
 /// are created to avoid race conditions
 pub fn set_monitor(file: File) -> Result<(), std::io::Error> {
+    // SAFETY: access global variable
```

Comment:
It needs to be further clarified, as accessing a global mutable variable is why the code is unsafe. 

I think what makes it safe here is we manually "ensured" the gloabal variable (`MONITOR`) will be written only once and all reads will happen after this variable is written.

Reply:
I took a stab at this. Please review.